hack/gen-swagger-doc/deploy.sh

```diff
@@ -5,24 +5,44 @@ set -e
 GITHUB_FQDN=github.com
 API_REF_REPO=kubevirt-incubator/api-reference
 API_REF_DIR=/tmp/api-reference
+GITHUB_IO_FQDN="https://kubevirt-incubator.github.io/api-reference"
+
+TARGET_DIR="$TRAVIS_BRANCH"
+if [ -n "${TRAVIS_TAG}" ] ; then
+    TARGET_DIR="$TRAVIS_TAG"
+fi
 
 git clone \
     "https://${API_REFERENCE_PUSH_TOKEN}@${GITHUB_FQDN}/${API_REF_REPO}.git" \
     "${API_REF_DIR}" >/dev/null 2>&1
-rm -rf "${API_REF_DIR}/content/"*
-cp -f _out/apidocs/html/*.html "${API_REF_DIR}/content/"
+rm -rf "${API_REF_DIR}/${TARGET_DIR:?}/"*
+cp -f _out/apidocs/html/*.html "${API_REF_DIR}/${TARGET_DIR}/"
 
 cd "${API_REF_DIR}"
 
+# Generate README.md file
+cat > README.md << __EOF__
+# KubeVirt API Reference
+
+Content of this repository is generated from OpenAPI specification of
+[KubeVirt project](https://github.com/kubevirt/kubevirt) .
+
+## KubeVirt API References
+
+* [master](${GITHUB_IO_FQDN}/master/index.html)
```

Comment:
One last question: would we get two master links if the master directory then exists already?

Reply:
I think we will not, there is regex `"^v[0-9.]*"` which matches directories starting with `v`.
But I am open to other suggestions to make less error prone!

Reply:
no change necessary. works for me. Let's see what happens.